src/charm.py

```diff
@@ -110,13 +140,14 @@ def _reconcile(self) -> None:
             RuntimeError: when initializing the lightkube client fails,
             or when creating the gateway resource fails.
         """
-        config = CharmConfig.from_charm(self)
+        client = get_client(self.app.name, self.model.name)
+        config = CharmConfig.from_charm(self, client)
         gateway_resource_definition = GatewayResourceDefinition.from_charm(self)
         _ = TLSInformation.from_charm(self)
```

Comment:
Why is the call needed if the argument is not used? Are we relying on side effects here?

Reply:
Yes, we use the `validate_config_and_integration` decorator to catch errors raised by `TLSInformation.from_charm` if the tls relation is not ready and set the charm to blocked state when that happens. It is defined [here](https://github.com/canonical/gateway-api-integrator-operator/blob/1c46079ae141ee3f891cac03a37ca2994a372289/src/state/validation.py#L21)

Reply:
Hm, is there another way to achieve the same goal in a more explicit way to improve code readability? It seems you must have quite some internal knowledge of the code base to understand this line.